doc/develop/index.rst

```diff
@@ -15,6 +15,7 @@ This document covers the main aspects of development with Odemis.
 .. toctree::
     :maxdepth: 2
 
+    coding-guidelines
     sys-arch
     extending
```

Comment:
I don't think it's nice for someone who want to starts with learning about Odemis to be bothered by every details of coding style. Let's place it after "extending".